### DISCUSSION TITLE
Ableton 12 public beta

### DISCUSSION THREAD

**Discussion by chozilla** [leolabs/ableton-js#113] [General]
Hi leo,

did you already sign up and had an opportunity to test the new Live 12 beta?
https://www.ableton.com/en/beta/

I have the feeling the connection tends to time out more easy. 
Maybe the program is less responsive on the socket?

I am also curious about the api... If we got any more possibilities with the new browser, new window management, and finally I want to see if we can set and delete cue_points >_<

Do you know how to run the python discovery code for the documentation of the API? I have not seen a V12 there.
Let me know if I can help with anything.

Best regards
\ chozilla

**Comment by leolabs** [comment]
I've tested ableton-js with Live 12 as well, but the connection seemed as stable as with Live 11 on my end. So far, these are the new functions I could find in Live 12's API:

- `Live.Application.Application.get_build_id()` Returns a string identifying the build.
- `Live.Application.Application.get_variant()` Returns one of the strings in Live.Application.Variants.
- `Live.Application.Application.show_message()` Shows a message box, returning the position of the pressed button.
- `Live.Application.Application.show_on_the_fly_message()` Same as show_message, but for when there is no predefined Text object.
- `Live.Application.Application.number_of_push_apps_running` Returns the number of connected Push apps.
- `Live.Base.get_text()` Retrieves the (translated) Text identified by `classname` and `textname`.
- `Live.CcControlDevice` CC Control Device
- `Live.MeldDevice` Meld Instrument Device
- `Live.RoarDevice` Roar Effect Device
- `Live.Song.Song.tuning_system` Access the currently active tuning system.

So no new ways to access `cue_points` and no new APIs for the browser, unfortunately.

**Reply by jbflow** [reply]
Yeah agreed it's a shame none of the browser stuff seems to have turned up in the API. It seems that because Push has moved away from using the scripts and now uses other technologies (from doing some digging I think it's using qt qml and maybe some OSC or direct MIDI stuff going on I'm not sure as I don't own Push 3). So the Python API might not get that much love any more as it's main purpose originally was to facilitate push and I think even Push 2's can be used without a script now.

@chozilla [here](https://github.com/NSUSpray/LiveAPI_MakeDoc) is the LiveAPI_MakeDoc script. I tried running it against Live 12 and ran in to some issues, but maybe one of you guys can figure it out 💡

**Reply by leolabs** [reply]
Hey @jbflow, I've been keeping a history of API docs on my machine. Here's a link to the docs I have so far: https://ableton-live-docs.vercel.app/

And here's a link to v12.0.2: https://ableton-live-docs.vercel.app/Live%2012.0.2.xml

**Reply by jbflow** [reply]
Ah Brilliant, thank you.

**Comment by jbflow** [comment]
@leolabs Another resource I found which is helpful is the decompiled scripts for live 12 [here](https://github.com/MrMatch246/AbletonLive12_MIDIRemoteScripts)
